From 9c8a57752f836306c4a30124f97b32f9c9a9448b Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:02 -0400
Subject: [ethernet] i40e: use stored base_queue value

Message-id: <1409658042-22689-182-git-send-email-sassmann@redhat.com>
Patchwork-id: 90373
O-Subject: [RHEL7.1 PATCH 181/281] i40e: use stored base_queue value
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

No need to read the PCI register for the PF's base queue on every single Tx
queue enable and disable as we already have the value stored from reading
the capability features at startup.

Change-ID: Ic02fb622757742f43cb8269369c3d972d4f66555
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit dfb699f970b24c47e526ccdecc460f7b88441f7c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_common.c b/drivers/net/ethernet/intel/i40e/i40e_common.c
index 8e4b33c..6e65f19 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_common.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_common.c
@@ -665,10 +665,9 @@ i40e_status i40e_get_mac_addr(struct i40e_hw *hw, u8 *mac_addr)
  **/
 void i40e_pre_tx_queue_cfg(struct i40e_hw *hw, u32 queue, bool enable)
 {
- u32 reg_val = rd32(hw, I40E_PFLAN_QALLOC);
- u32 first_queue = (reg_val & I40E_PFLAN_QALLOC_FIRSTQ_MASK);
- u32 abs_queue_idx = first_queue + queue;
+ u32 abs_queue_idx = hw->func_caps.base_queue + queue;
  u32 reg_block = 0;
+ u32 reg_val;
 
  if (abs_queue_idx >= 128)
   reg_block = abs_queue_idx / 128;
-- 
1.7.1